java/src/main/java/com/google/appengine/tools/pipeline/PipelineServiceFactory.java

```diff
@@ -14,18 +14,65 @@
 
 package com.google.appengine.tools.pipeline;
 
+import com.google.appengine.api.utils.SystemProperty;
 import com.google.appengine.tools.pipeline.impl.PipelineServiceImpl;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.datastore.Datastore;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.Base64;
 
 /**
  * A factory for obtaining instances of {@link PipelineService}
- * 
+ *
  * @author rudominer@google.com (Mitch Rudominer)
  */
 public final class PipelineServiceFactory {
   private PipelineServiceFactory() {
   }
 
+  /**
+   * @return PipelineService using application defaults for GCP infra needed underneath
+   */
+  @Deprecated //coupled to AppEngine
   public static PipelineService newPipelineService() {
-    return new PipelineServiceImpl();
+    return newPipelineService(SystemProperty.applicationId.get());
+  }
+
+  /**
+   * @param projectId GCP project under which pipelines will execute
+   * @return PipelineService that will execute pipelines in specific project, auth'd by application default credentials
+   */
+  public static PipelineService newPipelineService(String projectId) {
+    try {
+      GoogleCredentials credentials = GoogleCredentials.getApplicationDefault();
+      return new PipelineServiceImpl(projectId, credentials);
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to get default credentials", e);
+    }
+  }
+
+  public static PipelineService newPipelineService(String projectId, Datastore datastore) {
```

Comment:
this is what we need for testing; alternatively, we could pass `DatastoreOptions`, perhaps? Later we'll probably need more about the queues, as well.